src/com/lftechnology/java/training/srijan/basics/coreapp/ui/EmployeeView.java

```diff
@@ -0,0 +1,397 @@
+	public void adminAccess(Scanner input) {
+
+		Boolean looping = true;
+		do {
+			int userInput = 0;
+			LOGGER.log(
+				Level.INFO,
+				"\n1.Add User\n2.Search User \n3.Terminate User\n4.View Employee\n5.Exit");
+			try {
+				input = new Scanner(System.in);
+				userInput = input.nextInt();
+				if (userInput == 1) {
+					addUser(input);
+				}
+				else if (userInput == 2) {
+					searchUser(input);
+				}
+				else if (userInput == 3) {
+					terminateUser(input);
+				}
+				else if (userInput == 4) {
+					viewUsers();
+				}
+				else if (userInput == 5) {
+					exit();
+				}
+				else {
+					defaultUserView();
+				}
+			}
+			catch (Exception ex) {
+				LOGGER.log(Level.INFO, "Error:{0}", ex);
+			}
+		}
+		while (looping.equals(true));
+	}
+
+	/**
+	 * <p>Calls employee controller for employee add function</p>
+	 * @author srijan
+	 * @param input
+	 */
+	public void addUser(Scanner input) {
+
+		Employee employeeDetails = inputUserInfo(e, input);
+		employeeControl.addEmployee(employeeDetails);
+	}
+
+	/**
+	 * <p>Search user according to Department,fullname,address</p>
+	 * @param input
+	 * @author srijan
+	 */
+	public void searchUser(Scanner input) {
+
+		SearchEmployee emp = new SearchEmployee();
+		LOGGER.log(
+			Level.INFO,
+			"Search By:\n1.Department\n2.Fullname\n3.Address\n4.Default");
+		emp = choiceOfSearch(input);
+		if ("default".equals(emp.getSearchChoice())) {
+			employees = employeeControl.viewAllUser(e);
+			LOGGER.log(Level.INFO, "{0}", employees);
+		}
+		else {
+			employees = employeeControl.search(emp);
+			LOGGER.log(Level.INFO, "{0}", employees);
+		}
+	}
+
+	/**
+	 * <p>Terminates/Delete user from database and hence cannot login</p>
+	 * @author srijan
+	 */
+	public void terminateUser(Scanner input) {
+
+		employees = employeeControl.viewAllUser(e);
+		LOGGER.log(Level.INFO, "Employee Details\n{0}", employees);
+		LOGGER.log(
+			Level.INFO, "Choose id of an Employee you want to terminate");
+		int terminateEmployee = editAndTerminateEmployee(input);
+		employeeControl.terminate(terminateEmployee);
+	}
+
+	/**
+	 * <p>Displays all user/employees along with their information</p>
+	 * @author srijan
+	 */
+	public void viewUsers() {
+
+		employees = employeeControl.viewAllUser(e);
+		LOGGER.log(Level.INFO, "{0}", employees);
+	}
+
+	/**
+	 * <p>Exits to login view</p>
+	 * @author srijan
+	 */
+	public void exit() {
+
+		LOGGER.log(Level.INFO, "Logout successful");
+		LoginView.login();
+	}
+
+	/**
+	 * <p>If user input is incorrect then default shows all employees with their information</p>
+	 * @author srijan
+	 */
+	public void defaultUserView() {
+
+		employees = employeeControl.viewAllUser(e);
+		LOGGER.log(Level.INFO, "{0}", employees);
+	}
+
+	public void normalAccess(Scanner input, Employee e) {
+
+		Boolean loop = true;
+		do {
+			int userInput = 0;
+			LOGGER.log(
+				Level.INFO,
+				"\n1.View All User\n2.Edit Personal Details\n3.Exit");
+			try {
+				input = new Scanner(System.in);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "input". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)